Refactor: Update MovioVerticalCard and ForYouAndExploreScreen

baraa0abd opened the pull request:
<img width="1324" height="44" alt="image" src="https://github.com/user-attachments/assets/2f6e6d86-4e88-41e0-8b29-b397e246f543" />
<img width="1316" height="63" alt="image" src="https://github.com/user-attachments/assets/e1b1f4c1-77c8-4906-9e15-e850d9347051" />
<img width="1369" height="97" alt="image" src="https://github.com/user-attachments/assets/2e23fb7b-329c-4e29-a8e2-040209132dfb" />

Actor name text length in Movie and Series details screens
18- in for you section movie poster card's title has max 1 when in figma is 2 
19- vertical padding between posters not exists 
20- also poster card's rating has 4 digits
bug 15- not expected screen while loading search screen data when navigates from home to search screen
Remove matrial dep from presentation

baraa0abd committed:
Refactor: Remove empty search layout from ForYouAndExploreScreen

This commit removes the explicit empty search layout handling within `ForYouAndExploreScreen.kt`. The logic to display an image when `forYouMovies` is empty has been removed. The parent composable or a higher-level component is now expected to handle the empty state.

baraa0abd committed:
Refactor: Update MovioVerticalCard and ForYouAndExploreScreen

This commit introduces changes to the `MovioVerticalCard` and `ForYouAndExploreScreen` composables.

**MovioVerticalCard:**
- Removed the `paddingValue` parameter as it's no longer used.
- Ensured the `description` text handles long strings by truncating with an ellipsis and limiting to a single line.

**ForYouAndExploreScreen:**
- Removed the top padding from the `Column` composables that display loading and error states for a more consistent layout.
- Added bottom padding to the `LazyRow` displaying "For You" movies.
- Removed explicit height setting from the "For You" movies `LazyRow`.

baraa0abd committed:
Fix: Prevent long description overflow in MovioVerticalCard preview

The description text in the `VerticalCardPreview` has been shortened to prevent it from overflowing and causing visual issues in the preview.

baraa0abd committed:
Fix: Padding in ForYouAndExploreScreen

This commit adds padding to the `LazyRow` in `ForYouAndExploreScreen.kt`.

Specifically, it adds a 12dp padding around the `LazyRow` within the `ForYouScreen` composable.

baraa0abd committed:
Merge branch 'refs/heads/develop' into fix/18-movie-title-max-lines-incorrect-

baraa0abd committed:
Refactor: Update RateIcon and remove unused files

- Updated `RateIcon` to format the rate to one decimal place by truncating, not rounding.
- Removed unused commented-out code from `MovioEpisodesCard.kt`.
- Deleted `movieRow.kt` as it's no longer used.

baraa0abd committed:
Merge branch 'Issue/15-unexpected-screen-when-loading-search' into fix/18-movie-title-max-lines-incorrect-

# Conflicts:
#	presentation/src/main/java/com/madrid/presentation/screens/searchScreen/features/recentSearchLayout/ForYouAndExploreScreen.kt

baraa0abd committed:
Refactor: Remove empty state for "For You" movies

This commit removes the empty state handling from the "For You" movies section in `ForYouAndExploreScreen.kt`. The UI no longer displays an image when `forYouMovies` is empty.

baraa0abd committed:
Refactor: Format movie ratings to one decimal place

This commit introduces a utility function `toOneDecimalPlace` for `Double` values.
This function is used in `SimilarMoviesScreen.kt` to format the movie rating displayed in the `MovieCard`, ensuring it's shown with a single decimal place.
A try-catch block is included in the new function to handle potential exceptions during formatting, falling back to the original string representation if an error occurs.

baraa0abd committed:
Refactor: Enhance TopCastSection UI and text handling

This commit introduces several UI improvements to the `TopCastSection` composable:

- **Padding and Spacing:**
    - Added horizontal padding of `16.dp` to the main row containing the title and "See all" button.
    - Increased vertical spacing between the title row and the cast list to `16.dp`.
    - Added horizontal content padding of `16.dp` to the `LazyRow` for cast members.
    - Increased vertical spacing between the cast member image and name to `8.dp`.

- **Cast Member Item:**
    - Set a fixed width of `80.dp` for `CastMemberItem` to ensure consistent spacing between items.
    - Updated the text style for the cast member name to `Theme.textStyle.label.smallRegular12` for a better fit.
    - Enabled `maxLines = 2` and `overflow = TextOverflow.Ellipsis` for cast member names to handle long names gracefully.
    - Set `textAlign = TextAlign.Center` for cast member names.
    - Applied `modifier = Modifier.fillMaxWidth()` to the cast member name `MovioText` to allow text to wrap within the fixed width.

- **Preview:**
    - Updated the `TopCastSectionPreview` to include cast members with longer names and different name lengths to better test the text handling changes.

baraa0abd committed:
Fix: Adjust bottom padding in ForYouAndExploreScreen

The bottom padding for the LazyRow in `ForYouAndExploreScreen.kt` has been reduced from `32.dp` to `10.dp`.

baraa0abd committed:
Fix: Adjust bottom padding in ForYouAndExploreScreen

The bottom padding for the LazyRow in `ForYouAndExploreScreen.kt` has been reduced from `32.dp` to `10.dp`.

baraa0abd committed:
Merge remote-tracking branch 'origin/develop' into fix/18-movie-title-max-lines-incorrect-

# Conflicts:
#	presentation/src/main/java/com/madrid/presentation/screens/searchScreen/features/recentSearchLayout/ForYouAndExploreScreen.kt

baraa0abd committed:
Refactor: Update TopCastSection UI

Increased the size of the cast member image in `TopCastSection` from `60.dp` to `86.dp x 87.dp`.

ahmedhgabr commented:
![Screenshot_20250729_225224_Movio.jpg](https://github.com/user-attachments/assets/87215f80-b4d9-4c4a-a6a4-771da08b0dca)

- You get `10.` in case of 10.
- I guess we can move the logic from the component to another class (maybe try the view model)

baraa0abd committed:
Refactor: Remove Bottom Sheet components and update dependencies

This commit removes the following Bottom Sheet components from the `presentation` module:
- `MovieListsBottomSheet.kt`
- `RatingMovieBottomSheet.kt`
- `SaveMovieBottomSheet.kt`
- `ShareMovieBottomSheet.kt`

**Dependency Updates:**
- Removed `androidx.material3` and `androidx.material3.android` dependencies from `presentation/build.gradle.kts` as they are no longer directly used after the bottom sheet removals.

**Component Relocation:**
- Moved `ExpandableDescription.kt` from `presentation/screens/detailsScreen/componant` to `designSystem/component`.

**UI Updates:**
- Replaced `Text` with `MovioText` in `ForYouAndExploreScreen.kt` and `TrendingLayout.kt` for consistency.
- Updated `MovioButton.kt` in `designSystem` to:
    - Add an optional `backgroundColor` parameter. If `null`, the button will have no background and minimal padding, suitable for text buttons.
    - Introduce a new `MovioTextButton` composable for creating text-only buttons.

**Impact:**
- The removed Bottom Sheet components were specific to the `detailsScreen` and are no longer part of the application.
- The `ExpandableDescription` component is now part of the `designSystem` for broader reusability.
- UI components in search and home screens now use the consistent `MovioText` component.
- `MovioButton` offers more flexibility in styling, particularly for text buttons.

baraa0abd committed:
Refactor: Format rate display

This commit introduces a `RateFormatter` utility to consistently format media ratings across the application.

**Changes:**

- Created `RateFormatter.kt` with functions to format `Double` and `String` rates.
    - The formatter truncates the rate to one decimal place.
    - If the rate is a whole number after truncation, it's displayed as an integer.
- Integrated `RateFormatter` in:
    - `DetailsMovieViewModel`
    - `SeriesDetailsViewModel`
    - `UiStateMapper` (for movie and series mapping)
- Removed redundant rate formatting logic from `RateIcon.kt` and `SimilarMoviesScreen.kt` as it's now handled by `RateFormatter`.
- `SimilarMovie` data class in `SimilarMoviesScreen.kt` now expects a `String` for `rating`.
- Removed an unused log statement from `SeriesDetailsScreen.kt`.

baraa0abd committed:
Merge branch 'develop' into fix/18-movie-title-max-lines-incorrect-

fatimafaisalghazi commented on presentation/src/main/java/com/madrid/presentation/component/TopCastSection.kt:
```diff
@@ -121,10 +134,15 @@ fun TopCastSectionPreview() {
                 )
             )
         )
     }
```
i already fix the top cast section in my pr
